src/main/java/com/fundynamic/d2tm/game/map/MapRenderer.java

```diff
@@ -29,8 +33,35 @@ private void renderMap(Graphics graphics, Map map) throws SlickException {
         for (int x = 1; x <= map.getWidth(); x++) {
             for (int y = 1; y <= map.getHeight(); y++) {
                 Cell cell = map.getCell(x, y);
-                graphics.drawImage(cell.getTileImage(), (x - 1) * tileWidth, (y - 1) * tileHeight);
+                int cellX = (x - 1) * tileWidth;
+                int cellY = (y - 1) * tileHeight;
+                graphics.drawImage(cell.getTileImage(), cellX, cellY);
+
+                ShroudFacing shroudFacing = determineFacing(map, x, y);
+                if (shroudFacing != null) {
```

Comment:
perhaps have some null object here? 

I think this already shows how problematic drawing/rendering becomes. (like you said in the PR). We should have a chat about this.

The way I see it - there are several layers. And perhaps we need some kind of culling/preselecting stuff first, ordering, and then render all at once. We probably need to cull cells anyway, since we can't use the above approach with very big maps (which I want to support) - see https://github.com/stefanhendriks/dune2themaker4j/issues/15